### PR TITLE
Re-enable some Gen 2 test suites

### PR DESCRIPTION
**Pull request by shawyu-okta**
## Description:

To help prepare for bringing `master` into the gen 3 branch we want to re-enable some v2 test suites that we've been ignoring

## PR Checklist

- [ ] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-XXXXXX](https://oktainc.atlassian.net/browse/OKTA-XXXXXX)

### CODE REVIEW

File: playground/mocks/data/idp/idx/authenticator-enroll-password.json [hunk 1]
```diff
@@ -25,8 +25,7 @@
                   "secret": true
                 }
               ]
-            },
-            "required": true
```
Comment (shawyu-okta) [inline playground/mocks/data/idp/idx/authenticator-enroll-password.json:29]: This property is duplicate (see a few lines up) and causing lint failures

File: src/v3/jest.setup.js [hunk 2]
```diff
@@ -30,3 +31,13 @@ global.COMMITHASH = 'b9bbc0140703c3fbf0e2e58920362e70'; // "echo jest | md5"
...
+
+jest.mock('okta', () => {
+  return {
+    ...jest.requireActual('okta'),
+    loc: jest.fn().mockImplementation(
+      // eslint-disable-next-line no-unused-vars
+      (key, bundle, params) => (mockBundles.login[key] ? key : new Error(`Invalid i18n key: ${key}`)),
+    ),
+  };
+});
```
Comment (shawyu-okta) [inline src/v3/jest.setup.js:43]: This mock moved out of `__mocks__/okta.js` to be inlined here in the setup file. The other mock strategy was affecting v2 tests and mocking the `loc` function there as well.

File: test/unit/spec/v2/view-builder/views/__snapshots__/AutoRedirectView_spec.js.snap [hunk 3]
```diff
@@ -36,6 +36,7 @@ exports[`v2/view-builder/views/AutoRedirectView renders user identifier in title
         <div
           class="o-form-error-container"
           data-se="o-form-error-container"
+          role="alert"
```
Comment (shawyu-okta) [inline test/unit/spec/v2/view-builder/views/__snapshots__/AutoRedirectView_spec.js.snap:39]: I believe these snapshot changes come from a new version of courage.

File: test/types/api.test-d.ts [hunk 4]
```diff
@@ -35,16 +35,16 @@ expectType<void>(signIn.remove());
...
 signIn.on('afterError', (context: EventContext, error: EventErrorContext) => {
-  expectType<string>(context.controller);
+  expectType<string | null>(context.controller);
   expectType<string | undefined>(error.message);
   expectType<number | undefined>(error.statusCode);
   expectType<number | undefined>(error.xhr?.status);
 });
 signIn.on('afterRender', (context: EventContext) => {
-  expectType<string>(context.controller);
+  expectType<string | null>(context.controller);
```
Comment (shawyu-okta) [inline test/types/api.test-d.ts:47]: I updated this type to be more accurate before, just now updating this type test

File: .bacon.yml [hunk 5]
```diff
@@ -92,21 +92,20 @@ test_suites:
       sort_order: '13'
       timeout: '15'
       script_name: tsd
-      criteria: OPTIONAL
+      criteria: MERGE
       queue_name: small
-    # FIXME: revert criteria to "MERGE" when merging 0.0 to master
+      # FIXME: revert criteria to "MERGE" when merging 0.0 to master
```
Comment (glenfannin-okta) [inline .bacon.yml:97]: Will we do this later? Is there a ticket to track it?

**Review by glenfannin-okta — APPROVED** [review]

File: .bacon.yml (hunk 5 above)
Comment (shawyu-okta) [reply .bacon.yml:97]: Yeah I am going to re-enable this when I merge master and 0.1 together, which will be in this same ticket, actually. I just wanted to keep this PR small

**Pull request merged**